src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number | string;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: string,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+    public shape: string = 'triangle',
+  ) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('Arror');
+    }
+
+    if (
+      sideA + sideB <= sideC ||
+      sideA + sideC <= sideB ||
+      sideB + sideC <= sideA
+    ) {
+      throw new Error(
+        `throws an arror: ${sideA} and ${sideB} and ${sideC} cannot create triangle`,
+      );
+    }
+  }
+
+  getArea(): number | string {
+    const s = (this.sideA + this.sideB + this.sideC) / 2;
+
+    const result = Math.sqrt(
+      s * (s - this.sideA) * (s - this.sideB) * (s - this.sideC),
+    );
+
+    return Number.isInteger(result) ? result : Math.trunc(result * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  constructor(
+    public color: string,
+    public radius: number,
+    public shape: string = 'circle',
```

Comment:
[CODE KNOWLEDGE] - The 'shape' property is being unnecessarily assigned in the constructor. Since 'shape' is declared with an access modifier, TypeScript will automatically assign it.